src/common.c

```diff
@@ -220,18 +234,18 @@ int ofi_check_rx_mode(const struct fi_info *info, uint64_t flags)
 
 uint64_t fi_gettime_ms(void)
 {
-	struct timeval now;
+	struct timespec now;
 
-	gettimeofday(&now, NULL);
-	return now.tv_sec * 1000 + now.tv_usec / 1000;
+	clock_gettime(CLOCK_MONOTONIC, &now);
+	return now.tv_nsec / 1.0e6;
```

Comment:
This should be tv_sec * 1000 + tv_nsec / 1000000... Or:
```
fi_gettime_us() / 1000
```

Reply:
As an alternative -- following @shefty 's example above, 

```
fi_gettime_ns() {
    return now.tv_sec * 1000000000 + now.tv_nsec;
}

fi_gettime_ms {
    return fi_gettime_ns() / 1000000;
}

fi_gettime_us {
    return fi_gettime_ns() / 1000;
} 
```

Either way works, and I think it would be ideal to only make the call to clock_gettime in a single function so that if we need to modify the method of getting time in the future, we only need to modify it in one place.

Reply:
I would go with @jswaro 's option.  Also, please change the prefix from fi to ofi -- fi_gettime_xxx to ofi_gettime_xxx.  The fi prefix should be reserved for user visible APIs and data structures.  We just haven't gotten around to replacing it everywhere yet.